server/tests/oauth.test.js

```diff
@@ -0,0 +1,112 @@
+import data from './mochData/socialMockData';
+import VerifyUser from '../helpers/verifyUser';
+import CheckUser from '../middlewares/checkUser';
+
+import app from '../index';
+
+chai.use(sinonChai);
+chai.should();
+const { expect } = chai;
+
+
+describe('Social authentication tests', () => {
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should return user data when it get the profile from the social account', async () => {
+    const accessToken = 'zzzz-zzzz-zzzz';
+    const refreshToken = 'zzzzzzz';
+    const callBack = sinon.spy();
+    await VerifyUser.inSignSocial(accessToken, refreshToken, googleProfile, callBack);
+    expect(callBack.withArgs(null, googleProfile));
+  });
+});
+
+describe('Social authentication tests', () => {
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should return user data when it get the profile from the social account', async () => {
+    const accessToken = 'zzzz-zzzz-zzzz';
+    const refreshToken = 'zzzzzzz';
+    const callBack = sinon.spy();
+    await VerifyUser.inSignSocial(accessToken, refreshToken, facebookProfile, callBack);
+    expect(callBack.withArgs(null, facebookProfile));
+  });
+});
+
+describe('User sign in successfully with google', () => {
+  before(async () => {
+    sinon.stub(passport, 'authenticate').callsFake((strategy, options, callback) => {
+      callback(null, data[3], null);
+      return () => { };
+    });
+  });
+  after(() => {
+    passport.authenticate.restore();
+  });
+  it('should sign in user', (done) => {
+    request(app).get('/api/v1/auth/google/return')
+      .send(data[2])
+      .expect(200)
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        expect(res.body).to.be.a('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('message');
+        done();
+      });
+  });
+});
+
+
+describe('User sign in successfully with facebook', () => {
+  before(async () => {
+    sinon.stub(passport, 'authenticate').callsFake((strategy, options, callback) => {
+      callback(null, data[3], null);
+      return () => { };
+    });
+  });
+  after(() => {
+    passport.authenticate.restore();
+  });
+  it('should sign in user', (done) => {
+    request(app).get('/api/v1/auth/facebook/return')
+      .send(data[2])
+      .expect(200)
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        expect(res.body).to.be.a('object');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('message');
+        done();
+      });
+  });
+});
+
+describe('Authentication fails to return user google profile', () => {
+  it('should call next(error) once', () => {
+    const nextSpy = sinon.spy();
+
+    CheckUser.isAuthWithGoogle({}, {}, nextSpy);
+    expect(nextSpy.calledOnce).to.equal(true);
+  });
+});
+
+
+describe('Authentication fails to return user facebook profile', () => {
```

Comment:
Update the description to **should redirect the user to the Facebook Auth endpoint once not already authenticated**

Reply:
Feedback Implemented!